### DISCUSSION TITLE
How milvus able to work with dates?

### DISCUSSION THREAD

**Discussion by JenuelDev** [milvus-io/milvus#34676] [Q&A and General discussion]
we have added context with embeddings to milvus db. Now I don't know if this work but what if a user asked some about dates like. What are the best notes created last week?

lets say for example we had something like this in milvus.
```python
docs = [
    "James has called mike June 17, 2024",
    "Their is a complain about internet last June 14, 2024",
    ...
]
```

And lets imagine that today is June 18, 20204. And I would ask like:
```python
question = "Who called yesterday?"
```

It should get context related to June 17, 2024.

**Comment by tedxu** [comment]
The `date` or `datetime` types are not supported by milvus yet. You may use INT32 to store a timestamp. To query last week, use predicate expressions like:

```
today = datetime.today()
lastweek = today - timedelta(weeks=1)
lastweekts = int(datetime.timestamp(lastweek))
res = collection.query(
  expr = f"created > {lastweekts}", 
  output_fields = ["note"]
)
```

**Comment by xiaofan-luan** [comment]
@tedxu 
thought this could be a good first issue to work with
